app/store.go

```diff
@@ -111,7 +111,7 @@ func (s *StoreApp) parseAppState(data []byte, chainID string, init weave.Initial
 	var appState weave.Options
 	err := json.Unmarshal(data, &appState)
 	if err != nil {
-		return errors.WithCode(err, errors.CodeTxParseError)
+		return errors.ErrInternal.New(err.Error())
```

Comment:
I would prefer

errors.Wrap(err, "json decoding")

It will get internal error code automatically, and seems simpler if we just want to return the raw error with ErrInternal